compiler/ast.nim

```diff
@@ -295,6 +295,9 @@ type
     sfUsedInFinallyOrExcept  # symbol is used inside an 'except' or 'finally'
     sfSingleUsedTemp  # For temporaries that we know will only be used once
     sfNoalias         # 'noalias' annotation, means C's 'restrict'
+    sfCompileToCpp
```

Comment:
I don't understand this. You might as well remove the ``sfCompileToCpp`` alias.

Reply:
before PR `sfCompileToCpp` and `sfInfixCall` were aliases, but after PR they're not aliases and I need the distinction to support C++ free functions, which fixes #12150; I've went through each use of sfCompileToCpp vs sfInfixCall to make sure the correct one is used in each case.